test/strings.jl

```diff
@@ -1280,20 +1280,77 @@ end
 @test isxdigit("a") == true
 @test isxdigit("g") == false
 
-@test is_valid_ascii("is_valid_ascii") == true
-@test is_valid_ascii("Σ_not_valid_ascii") == false
-@test is_valid_char('a') == true
-@test is_valid_char('\x00') == true
-@test is_valid_char('\ud800') == false
-
-@test is_valid_utf16(utf16("a")) == true
-@test is_valid_utf16(utf16("\ud800")) == false
-# TODO is_valid_utf8
-
 # Issue #11140
-@test is_valid_utf32(utf32("a")) == true
-@test is_valid_utf32(utf32("\x00")) == true
-@test is_valid_utf32(UInt32[0xd800,0]) == false
+@test isvalid(utf32("a")) == true
+@test isvalid(utf32("\x00")) == true
+@test isvalid(UTF32String, UInt32[0xd800,0]) == false
+
+# Issue #11241
+
+@test isvalid(ASCIIString, "is_valid_ascii") == true
+@test isvalid(ASCIIString, "Σ_not_valid_ascii") == false
+@test isvalid(Char, 'a') == true
+@test isvalid(Char, '\x00') == true
+
+# test all edge conditions
+@test isvalid(Char, 0) == true
```

Comment:
Yes, that’s mine too... I just followed what was already there...  and didn’t know all the ins-and-outs of how for works in Julia (like the trick of getting two values at once above)...
Can cleaning up the character & string testing though be put into a later PR?  (which of course I’ll do shortly)  Thanks!